### PR TITLE
Enter logic for Quadratic hashing

### PR DESCRIPTION
**Pull request by silam**
### Describe your change:

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/hashing/quadratic_probing.py [hunk 1]
```diff
@@ -1,3 +1,26 @@
+
```
Comment (CaedenPH) [inline data_structures/hashing/quadratic_probing.py:1]:
```suggestion
```

File: data_structures/hashing/quadratic_probing.py [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+The logic is
+
+For the loop of the array, running from 0 to array length, given x is an element of the array
+1 - Check if the slot hash(x) % SizeofHashtable is full, then we try (hash(x) + 1*1) % SizeofHashtable.
+2 - then check if (hash(x) + 1*1) % SizeofHashtable is also full, then we try (hash(x) + 2*2) % SizeofHashtable.
+3 - then check If (hash(x) + 2*2) % SizeofHashtable is also full, then we try (hash(x) + 3*3) % SizeofHashtable.
+...
+This process is repeated for all the values of i until an empty slot is found.
+
```
Comment (CaedenPH) [inline data_structures/hashing/quadratic_probing.py:22]:
```suggestion
```

File: data_structures/hashing/quadratic_probing.py [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+"""
 #!/usr/bin/env python3
```
Comment (CaedenPH) [inline data_structures/hashing/quadratic_probing.py:24]: You must move the shebang to the first line

**Review by CaedenPH — CHANGES_REQUESTED** [review]
The shebang must be on the very first line

File: data_structures/hashing/quadratic_probing.py (hunk 3 above)
Comment (silam) [reply data_structures/hashing/quadratic_probing.py]: Move #!/usr/bin/env python3 to the first line

File: data_structures/hashing/quadratic_probing.py (hunk 1 above)
Comment (silam) [reply data_structures/hashing/quadratic_probing.py]: Move #!/usr/bin/env python3 to the first line

**Review by CaedenPH — APPROVED** [review]
Looks good, thanks

**Comment by silam** [comment]
How do I know what I did that I got label: tests are failing

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Lines are > 88 characters per line.

**Pull request closed without merging**